dlt/destinations/dataset.py

```diff
@@ -318,3 +321,29 @@ def dataset(
     if dataset_type == "dbapi":
         return ReadableDBAPIDataset(destination, dataset_name, schema)
     raise NotImplementedError(f"Dataset of type {dataset_type} not implemented")
+
+
+def create_ibis_backend(
+    destination: TDestinationReferenceArg, dataset_name: str, schema: Schema
+) -> IbisBackend:
+    from dlt.common.libs.ibis import ibis
+
+    # TODO: abstract out destination client related stuff
+    destination = Destination.from_reference(destination)
+    client_spec = destination.spec()
+    if isinstance(client_spec, DestinationClientDwhConfiguration):
+        client_spec._bind_dataset_name(dataset_name=dataset_name, default_schema_name=schema.name)
+    client = destination.client(schema, client_spec)
+
+    if destination.destination_type not in [
+        "dlt.destinations.postgres",
+        "dlt.destinations.duckdb",
+        "dlt.destinations.filesystem",
+    ]:
+        raise NotImplementedError()
+
+    ibis = ibis.connect(client.config.credentials.to_native_representation())
+    # NOTE: there seems to be no standardized way to set the current dataset / schema in ibis
```

Comment:
I'm not sure where we should put the stuff that converts our credentials into something IBIS understands. `client.config.credentials.to_native_representation()` could work for many cases, but selecting the default dataset will probably be something destination specific..

Reply:
Also for the filesystem destination this will work quite differently, because we first need to populate the duckdb database and then attach ibis to it.

Reply:
1. part that creates backend should go to dlt/helpers/ibis
2. part that converts credentials should go to common/libs/ibis
here we can improve a lot. we already pollute credentials in common with concrete converting functions (like fsspec, delta etc.) we should create some universal mechanism to convert them.